test/built-ins/Temporal/PlainDateTime/prototype/withPlainDate/argument-string-iso-calendar.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.withplaindate
+description: New calendar is preserved if original PDT has ISO calendar
+features: [Temporal]
+includes: [temporalHelpers.js]
+---*/
+
+const dt = new Temporal.PlainDateTime(1995, 12, 7, 3, 24, 30);
+const cal = {
+  id: 'thisisnotiso',
+  era() { return "the era"; },
+  eraYear() { return 1909; },
+  toString() { return "this is a string"; },
+  year() { return 2008; },
+  month() { return 9; },
+  monthCode() { return "M09"; },
+  day() { return 6; }
+};
+const shifted = dt.withCalendar(cal).withPlainDate("2008-09-06");
```

Comment:
The argument passed here is the same as what your calendar is going to return anyway - let's use another argument.